### PR TITLE
Improve documentation about default argument behaviour

### LINKED ISSUE #59: check which optional attributes should have a default
This is a follow up of #38 

> For resource schema, if you have an optional attribute, typically (but not always) you would include a default.

### PR DESCRIPTION
**Pull request by psycofdj**
fixes #59

### CODE REVIEW

**Comment by gberche-orange** [comment]
@psycofdj sorry I did not notified you explicitly as part of #73 which included some fixes for #59

File: website/docs/r/buildpack.html.markdown [hunk 1]
```diff
@@ -30,7 +30,7 @@ resource "cf_buildpack" "tomee" {
 The following arguments are supported:
 
 * `name` - (Required) The name of the Buildpack.
-* `position` - (Optional, Number) Specifies where to place the buildpack in the detection priority list. For more information, see the [Buildpack Detection](https://docs.cloudfoundry.org/buildpacks/detection.html) topic.
+* `position` - (Optional, Number) Specifies where to place the buildpack in the detection priority list. For more information, see the [Buildpack Detection](https://docs.cloudfoundry.org/buildpacks/detection.html) topic. When not provided, value is computed and retreived from Cloud Foundry.
```
Comment (gberche-orange) [inline website/docs/r/buildpack.html.markdown]: suggesting: 

When not provided, cloudfoundry assigns a default buildpack position.

(I looked up Cf CLI docs and code and could not find code assigning default position. [CC API](http://apidocs.cloudfoundry.org/1.27.0/buildpacks/creates_an_admin_buildpack.html) does not detail it either default value. Would need to look into cloud controller tests)

File: website/docs/r/space.html.markdown [hunk 2]
```diff
@@ -46,12 +46,12 @@ The following arguments are supported:
 * `name` - (Required) The name of the Space in Cloud Foundry.
 * `org` - (Required) The ID of the [Org](/docs/providers/cloudfoundry/r/org.html) within which to create the space.
 * `quota` - (Optional) The ID of the Space [quota](/docs/providers/cloudfoundry/r/quota.html) or plan defined for the owning Org. By default, no space quota is assigned to the space.
```
Comment (gberche-orange) [inline website/docs/r/space.html.markdown]: Suggesting refinement

* `quota` - (Optional) The ID of the Space [quota](/docs/providers/cloudfoundry/r/quota.html) or plan defined for the owning Org. Specifying an empty string requests unassigns any space quota from the space. Defaults to empty string.

File: website/docs/r/app.html.markdown [hunk 3]
```diff
@@ -27,17 +27,18 @@ The following arguments are supported:
 
 * `name` - (Required) The name of the application.
 * `space` - (Required) The GUID of the associated Cloud Foundry space.
-* `instances` - (Optional, Number) The number of app instances to start.
-* `memory` - (Optional, Number) The memory limit for each application instance in megabytes.
-* `disk_quota` - (Optional, Number) The disk space to be allocated for each application instance in megabytes.
-* `stack` - (Optional) The GUID of the stack the application will be deployed to. Use the [`cf_stack`](/docs/providers/cf/d/stack.html) data resource to lookup the stack GUID to overriding the default.
+* `ports` - (Optional, Array of Number) A list of ports which the app will listen on. If not provided, value is computed and retreived from Cloud Foundry.
```
[the remark this answers is not in the document]
Comment (gberche-orange) [reply website/docs/r/app.html.markdown]: relates to #63 assigning it to @psycofdj

[the remark this answers is not in the document]
Comment (psycofdj) [reply website/docs/r/app.html.markdown]: There is a chance that the `port` feature will appear in `v3`. This api is still moving, even `service_instances` and `service_bindings` are still marked as experimental.
I suggest to only remove the `ports` attribute form the documentation and leave the resource as it is. @gberche-orange  ?

[the remark this answers is not in the document]
Comment (gberche-orange) [reply website/docs/r/app.html.markdown]: @psycofdj 

> I suggest to only remove the ports attribute form the documentation and leave the resource as it is.

I agree.

File: website/docs/r/asg.html.markdown [hunk 4]
```diff
@@ -49,9 +49,9 @@ The following arguments are supported:
   - `destination` - (Required, String) The IP address or CIDR block that can receive traffic.
   - `ports` - (Required, String) A single port, comma-separated ports or range of ports that can receive traffic.
```
Comment (gberche-orange) [inline website/docs/r/asg.html.markdown]: @psycofdj Please consider removing as discussed in https://github.com/mevansam/terraform-provider-cf/pull/76/files/5035d55096abebfa482ecfe19218ff23b2814ce4#r192446675

**Comment by gberche-orange** [comment]
@psycofdj can you please resolve conflicts, so that when tests are green we can merge the pr ?

**Comment by psycofdj** [comment]
@janosbinder Thanks, I don't understand how this line got edited on my commit...

**Comment by janosbinder** [comment]
@psycofdj almost good, we have some issues:

```--- FAIL: TestAccOrg_importBasic (5.45s)
	testing.go:513: Step 0 error: Error applying: 6 error(s) occurred:
		
		* cloudfoundry_user.u1: 1 error(s) occurred:
		
		* cloudfoundry_user.u1: user test-user1@acme.com already exists
		* cloudfoundry_quota.runaway: 1 error(s) occurred:
		
		* cloudfoundry_quota.runaway: Server error, status code: 400, error code: 240002, message: Quota Definition is taken: runaway_test
		* cloudfoundry_user.u5: 1 error(s) occurred:
		
		* cloudfoundry_user.u5: user test-user5@acme.com already exists
		* cloudfoundry_user.u4: 1 error(s) occurred:
		
		* cloudfoundry_user.u4: user test-user4@acme.com already exists
		* cloudfoundry_user.u3: 1 error(s) occurred:
		
		* cloudfoundry_user.u3: user test-user3@acme.com already exists
		* cloudfoundry_user.u2: 1 error(s) occurred:
		
		* cloudfoundry_user.u2: user test-user2@acme.com already exists```

Can you please clean up these artifacts and also include those steps in the `clean.sh`?

**Comment by psycofdj** [comment]
@janosbinder I added these resources in the clean script. Tests are better but we still have an issue with the mysql-p broker.

Could we merge the branch anyway ?

**Comment by janosbinder** [comment]
Thanks @psycofdj for the fix - please merge.

**Pull request merged**